mne_bids/tests/test_utils.py

```diff
@@ -124,3 +144,47 @@ def test_infer_eeg_placement_scheme():
     raw.rename_channels({'P3': 'foo'})
     placement_scheme = _infer_eeg_placement_scheme(raw)
     assert placement_scheme == 'n/a'
+
+
+def test_check_datatype():
+    """Test checking if datatype exists in raw data."""
+    sfreq, n_points = 1024., int(1e6)
+    rng = RandomState(99)
+    info_eeg = mne.create_info(['ch1', 'ch2', 'ch3'], sfreq, ['eeg'] * 3)
+    raw_eeg = mne.io.RawArray(rng.random((3, n_points)) * 1e-6, info_eeg)
+    info_meg = mne.create_info(['ch1', 'ch2', 'ch3'], sfreq, ['mag'] * 3)
+    raw_meg = mne.io.RawArray(rng.random((3, n_points)) * 1e-6, info_meg)
+    info_ieeg = mne.create_info(['ch1', 'ch2', 'ch3'], sfreq, ['seeg'] * 3)
+    raw_ieeg = mne.io.RawArray(rng.random((3, n_points)) * 1e-6, info_ieeg)
+    datatype = None
+    with pytest.raises(ValueError, match=f'The specified datatype {datatype} '
+                                         'is currently not supported.'):
+        _check_datatype(raw_eeg, datatype)
+    datatype = 'anat'
+    with pytest.raises(ValueError, match=f'The specified datatype {datatype} '
+                                         'is currently not supported.'):
+        _check_datatype(raw_eeg, datatype)
+    datatype = 'eeg'
+    _check_datatype(raw_eeg, datatype)
+    with pytest.raises(ValueError, match=f'The specified datatype {datatype} '
+                                         'was not found in the raw object.'):
+        _check_datatype(raw_meg, datatype)
```

Comment:
```suggestion
    for datatype, raw in [(None, raw_eeg, ('eeg', raw_meg), ('anat', raw_eeg)]:
        with pytest.raises(ValueError, match=f'The specified datatype {datatype} '
                                             'is currently not supported.'):
            _check_datatype(raw, datatype)
```

can you compress with something like this?

Reply:
I did, hope it is fine the way I have solved it now :)